doc/release/0.17.0-notes.rst

```diff
@@ -29,6 +29,11 @@ Deprecated features
 Backwards incompatible changes
 ==============================
 
+The functions ``trim1`` and ``trimboth`` now partially (for numpy >= 1.8.0) or
+fully (for numpy < 1.8.0) sort elements before trimming. This to make sure 
+the trimmed elements are the lowest or highest depending on the case. 
+Slicing without at least partial sorting was random hacking.
```

Comment:
I'd rephrase that a bit: "was random hacking" --> "was previously done, but didn't make sense for unsorted input"